### ISSUE TITLE
With runbot_gitlab module fails github repositories

### ISSUE THREAD

**Issue by angelmoya** [OCA/runbot-addons#35]
The module tray to set gitlab ci url, and fails, also if check gitlab is not set, for example in github repositories

```
2015-06-03 16:19:41,161 5269 ERROR runbot_domatix openerp.http: Exception during JSON request handling.
Traceback (most recent call last):
  File "/home/runbot/server/openerp/http.py", line 530, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/runbot/server/openerp/http.py", line 567, in dispatch
    result = self._call_function(**self.params)
  File "/home/runbot/server/openerp/http.py", line 303, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/runbot/server/openerp/service/model.py", line 113, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/runbot/server/openerp/http.py", line 300, in checked_call
    return self.endpoint(*a, **kw)
  File "/home/runbot/server/openerp/http.py", line 796, in __call__
    return self.method(*args, **kw)
  File "/home/runbot/server/openerp/http.py", line 396, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/server/addons/web/controllers/main.py", line 935, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/server/addons/web/controllers/main.py", line 927, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/home/runbot/server/openerp/api.py", line 241, in wrapper
    return old_api(self, *args, **kwargs)
  File "/home/runbot/server/openerp/api.py", line 363, in old_api
    result = method(recs, *args, **kwargs)
  File "/home/runbot/runbot-addons/runbot_gitlab/runbot_repo.py", line 149, in write
    self.id,
  File "/home/runbot/runbot-addons/runbot_gitlab/runbot_repo.py", line 111, in set_gitlab_ci_conf
    _('Gitlab repo requires an API token from a user with '
except_orm: (u'\xa1Error!', u'Gitlab repo requires an API token from a user with admin access to repo.')
```

**Comment by bwrsandman** [comment]
As the message says: you need an API key to be able to change the CI settings of the repo.

For gitlab: https://gitlab.savoirfairelinux.com/profile/account your user needs to be master of the project in question.

**Comment by angelmoya** [comment]
Ok, but error appears when edit a **github** repository, no in gitlab repository.

It would check if repository is gitlab before raise this exception... I will check code and prepare a PR.

**Comment by bwrsandman** [comment]
I see. Are you mixing gitlab and github repos in your runbot instance?
It is possible that incompatibility has been introduced with that module and it should be fixed.

**Comment by angelmoya** [comment]
Yes, I would like to config OCA repositories and gitlab repositories.

I will test all proces, thanks

**Comment by hbrunn** [comment]
fixed by #72
